superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx

```diff
@@ -102,22 +96,24 @@ const collapseStyles = (theme: SupersetTheme) => css`
   }
 `;
 
-export default function SqlEditorLeftBar({
-  actions,
+const SqlEditorLeftBar = ({
   database,
-  queryEditor,
+  queryEditorId,
   tables = [],
   height = 500,
   setEmptyState,
-}: SqlEditorLeftBarProps) {
+}: SqlEditorLeftBarProps) => {
+  const dispatch = useDispatch();
+  const queryEditor = useQueryEditor(queryEditorId, ['schema']);
+
   // Ref needed to avoid infinite rerenders on handlers
   // that require and modify the queryEditor
-  const queryEditorRef = useRef<QueryEditor>(queryEditor);
+  const queryEditorRef = useRef(queryEditor);
```

Comment:
`queryEditorRef` won't be no longer needed. Please use queryEditor (from line 107) instead.

Please remove the following block around line 135.

```
  useEffect(() => {
    queryEditorRef.current = queryEditor;
  }, [queryEditor, database]);```